sprout/appliances/tasks.py

```diff
@@ -1584,17 +1589,31 @@ def delete_template_from_provider(self, template_id):
 
 
 @singleton_task()
-def appliance_rename(self, appliance_id, new_name):
-    self.logger.info("Start renaming process for {} to {}".format(appliance_id, new_name))
+def appliance_rename(self, appliance_id):
     try:
         appliance = Appliance.objects.get(id=appliance_id)
+        new_name = '{}_'.format(appliance.appliance_pool.owner.username)
+        if appliance.version and not appliance.version.startswith('...'):
+            # CFME
+            new_name += 'cfme_{}_'.format(appliance.version.replace('.', ''))
+        else:
+            # MIQ
+            new_name += 'miq_'
+        new_name += '{}_{}'.format(
+            appliance.template.date.strftime("%y%m%d"),
+            fauxfactory.gen_alphanumeric(length=4))
+        self.logger.info("Start renaming process for {} to {}".format(appliance_id, new_name))
     except ObjectDoesNotExist:
+        self.logger.warning("No such appliance {} in sprout".format(appliance_id))
         return None
     if not appliance.provider.is_working:
         raise RuntimeError('Provider is not working.')
-    if not appliance.provider.allow_renaming:
+    if (not appliance.provider.allow_renaming or appliance.is_openshift or
```

Comment:
Looks like this could be checked immediately after pulling the appliance object, before making additional calls/queries to generate the new name